src/main/java/org/folio/service/orders/PurchaseOrderLineService.java

```diff
@@ -490,5 +499,26 @@ public Future<Void> validateAndNormalizeISBNCommon(List<CompositePoLine> composi
           .mapEmpty();
       });
   }
+
+  public Future<Void> updateSearchLocations(PoLine poLine, RequestContext requestContext) {
+    if (CollectionUtils.isEmpty(poLine.getLocations())) {
+      return Future.succeededFuture();
+    }
+
+    var holdingIds = StreamEx.of(poLine.getLocations()).map(Location::getHoldingId).nonNull().toList();
+    if (CollectionUtils.isEmpty(holdingIds)) {
```

Comment:
I don't think it's correct, it's possible to send list of locations with location ids only, so if there are no holding ids you still should combine a list

Reply:
but inventoryManager.getHoldingsByIds(holdingIds, requestContext) should be called only when there are holding ids

Reply:
nice catch. review it again please